mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java

```diff
@@ -222,134 +280,423 @@ public void testSimpleAttachment3() throws MessagingException, IOException {
+        
+        MimeBodyPart part = new MimeBodyPart(new ByteArrayInputStream(new byte[0]));
+        part.setFileName("removeMe.tmp");
+        multiPart.addBodyPart(part);
+        MimeMessage message = new MimeMessage(Session
+                .getDefaultInstance(new Properties()));
+        message.setContent(multiPart);
+        message.saveChanges();
+        Mail mail = mock(Mail.class);
+        //When
+        mailet.analyseMultipartPartMessage(message, mail);
+        //Then
+        assertThat(multiPart.getCount()).isZero();
+    }
 
+    @Test
+    public void saveAttachmentShouldUsePartNameIfNoFilename() throws Exception {
+        //Given
+        StripAttachment mailet = new StripAttachment();
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .setProperty("directory", folderPath)
+                .setProperty("pattern", ".*\\.tmp")
+                .build();
+        mailet.init(mci);
+        Part part = new MimeBodyPart(new ByteArrayInputStream(new byte[0]));
+        part.setFileName("example.tmp");
+        String fileName = null;
+        //When
+        String actual = mailet.saveAttachmentToFile(part, fileName);
+        //Then
+        assertThat(actual).startsWith("example");
+        assertThat(actual).endsWith(".tmp");
     }
+    
+    @Test
+    public void saveAttachmentShouldThrowWhenNoFilenameAtAll() throws Exception {
+        StripAttachment mailet = new StripAttachment();
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .setProperty("directory", folderPath)
+                .setProperty("pattern", ".*\\.tmp")
+                .build();
+        mailet.init(mci);
+        Part part = new MimeBodyPart(new ByteArrayInputStream(new byte[0]));
+        String fileName = null;
 
+        expectedException.expect(NullPointerException.class);
+        mailet.saveAttachmentToFile(part, fileName);
+    }
+    
+    @Test
+    public void saveAttachmentShouldAddBinExtensionWhenNoFileNameExtension() throws Exception {
+        //Given
+        StripAttachment mailet = new StripAttachment();
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .setProperty("directory", folderPath)
+                .setProperty("pattern", ".*")
+                .build();
+        mailet.init(mci);
+        Part part = new MimeBodyPart(new ByteArrayInputStream(new byte[0]));
+        String fileName = "exampleWithoutSuffix";
+        //When
+        String actual = mailet.saveAttachmentToFile(part, fileName);
+        //Then
+        assertThat(actual).startsWith("exampleWithoutSuffix");
+        assertThat(actual).endsWith(".bin");
+    }
+    
     private Mailet initMailet() throws MessagingException {
         Mailet mailet = new StripAttachment();
 
         FakeMailetConfig mci = FakeMailetConfig.builder()
                 .mailetName("Test")
-                .setProperty("directory", "./")
+                .setProperty("directory", folder.getRoot().getPath())
```

Comment:
Ok, I guess the problem is the following : you test with too much configuration option at once.

You should : 
 - 1/ Show us the minimal configuration to start the mailet
 - 2/ Then activate parameters one by one to show their effect (so that I can understand the effect of each one of them)

Actually, from the beginning of the Sprint, threw the reviews, I did not get how this mailet worked.